eth/vm/code_stream.py

```diff
@@ -25,34 +25,34 @@ class CodeStream(object):
 
     def __init__(self, code_bytes: bytes) -> None:
         validate_is_bytes(code_bytes, title="CodeStream bytes")
-        self.stream = io.BytesIO(code_bytes)
+        stream = io.BytesIO(code_bytes)
+        self.stream = stream
+        self._bound_stream_read = stream.read
         self._raw_code_bytes = code_bytes
         self._length_cache = len(code_bytes)
         self.invalid_positions = set()
         self.valid_positions = set()
 
     def read(self, size: int) -> bytes:
-        return self.stream.read(size)
+        return self._bound_stream_read(size)
 
     def __len__(self) -> int:
         return self._length_cache
 
     def __iter__(self) -> 'CodeStream':
         return self
 
-    def __next__(self) -> int:
-        return self._next()
-
     def __getitem__(self, i: int) -> int:
         return self._raw_code_bytes[i]
 
-    def _next(self) -> int:
-        next_opcode_as_byte = self.read(1)
+    def __next__(self) -> int:
```

Comment:
Maybe instead of having `CodeStream` implement a `__next__` we could just do this in the `__iter__` and make it a generator function that does this in a loop.  That way we don't have to do the `self._bound_stream_read` property lookup and it could also be faster by not having to create a new call stack every time assuming generators are able to re-use it?

Reply:
Huh, I'm quite surprised that a speedup is nearly undetectable from this change. It's within tolerance for variation from run to run (eg~ 2.6->2.4%). However, it's not worse, and I'm so convinced that it *should* be better that I'll go ahead and make the change.

Reply:
looks like this changed didn't make it in (just making a note of it)

Reply:
Ah, so `__next__` isn't getting called by the iterator anymore, but is still used by `.peek()` so I left it in.

A silly example to show that `__next__` isn't being used anymore:
```
In [1]: class IterMe:
   ...:     def __iter__(self):
   ...:         yield 1
   ...:         yield 2
   ...:     def __next__(self):
   ...:         return 'a'
   ...:     

In [3]: i = IterMe()

In [4]: next(i)
Out[4]: 'a'

In [5]: [v for v in i]
Out[5]: [1, 2]

Reply:
If it's only called by `peek` then how about in-lining it there?